openshift_cli_installer/utils/gcp_utils.py

```diff
@@ -0,0 +1,77 @@
+import json
+import os
+import shortuuid
+import shutil
+from pathlib import Path
+from simple_logger.logger import get_logger
+from openshift_cli_installer.utils.const import GCP_STR
+
+LOGGER = get_logger(name=__name__)
+short_uuid = shortuuid.uuid()
+
+
+def get_gcp_params():
+    gcp_params = {}
+    gcp_params["gcp_sa_file_dir"] = os.path.join(os.path.expanduser("~"), ".gcp")
+    gcp_params["openshift_installer_gcp_sa_file_path"] = os.path.join(
+        gcp_params["gcp_sa_file_dir"], "osServiceAccount.json"
+    )
+    gcp_params["backup_existing_gcp_sa_file_path"] = os.path.join("/tmp", f"installerTemp{short_uuid}.json")
+    return gcp_params
+
+
+def set_gcp_configuration(user_input):
+    """
+    Saves provided GCP Service Account file at location '~/.gcp/osServiceAccount.json'
+
+    If file already exists at this path, it will be copied to tmp file first.
+
+    """
+    if is_platform_gcp(user_input.clusters):
+        gcp_params = get_gcp_params()
+        if os.path.exists(gcp_params["openshift_installer_gcp_sa_file_path"]):
+            LOGGER.info(
+                f"File {gcp_params['openshift_installer_gcp_sa_file_path']} already exists. Copying to {gcp_params['backup_existing_gcp_sa_file_path']}"
+            )
+            shutil.copy(
+                gcp_params["openshift_installer_gcp_sa_file_path"], gcp_params["backup_existing_gcp_sa_file_path"]
+            )
+        else:
+            Path(gcp_params["gcp_sa_file_dir"]).mkdir(parents=True, exist_ok=True)
+        LOGGER.info(f"Saving GCP ServiceAccount file to {gcp_params['openshift_installer_gcp_sa_file_path']}")
+        shutil.copy(user_input.gcp_service_account_file, gcp_params["openshift_installer_gcp_sa_file_path"])
+
+
+def restore_gcp_configuration(user_input):
+    """
+    Restores location '~/.gcp/osServiceAccount.json'
+
+    Copy file from tmp location to '~/.gcp/osServiceAccount.json' if exists before,
+    Otherwise removes the directory '~/.gcp'
+
+    """
+    if is_platform_gcp(user_input.clusters):
+        gcp_params = get_gcp_params()
+        if os.path.exists(gcp_params["backup_existing_gcp_sa_file_path"]):
+            LOGGER.info(f"Restoring previous file contents of {gcp_params['openshift_installer_gcp_sa_file_path']}")
+            shutil.copy(
+                gcp_params["backup_existing_gcp_sa_file_path"], gcp_params["openshift_installer_gcp_sa_file_path"]
+            )
+            os.remove(gcp_params["backup_existing_gcp_sa_file_path"])
+        else:
+            LOGGER.info(f"Deleting path {gcp_params['openshift_installer_gcp_sa_file_path']}")
+            shutil.rmtree(gcp_params["gcp_sa_file_dir"])
+
+
+def is_platform_gcp(clusters):
```

Comment:
maybe `has_gcp_platform` or something like that is more compatible since we are looking across multiple clusters

Reply:
thanks for the suggestion, I would prefer to keep the function name something like `is_clusters_platform_gcp`, following the same funtion names as we have in other files.

Reply:
renamed function to `is_clusters_gcp_platform`